Dockerfile.fpm

```diff
@@ -1,27 +1,23 @@
+ARG composer_type=no-dev
+ARG image_tag=latest
+FROM elifesciences/annotations_composer:${image_tag}-${composer_type} AS build
 FROM elifesciences/php_fpm:d83fc4714914898b0842199578e1cc88d9feab2a
 
-USER root
-RUN docker-php-ext-install pcntl
-
-USER elife
 ENV PROJECT_FOLDER=/srv/annotations
 ENV PHP_ENTRYPOINT=web/app.php
-RUN mkdir ${PROJECT_FOLDER}
-WORKDIR /srv/annotations
-COPY --chown=elife:elife composer.json composer.lock /srv/annotations/
-RUN composer-install
-
-# yes this is how you copy directories
-COPY --chown=elife:elife bin/ /srv/annotations/bin/
-COPY --chown=elife:elife config/ /srv/annotations/config/
-COPY --chown=elife:elife src/ /srv/annotations/src/
-COPY --chown=elife:elife web/ /srv/annotations/web/
-COPY --chown=elife:elife smoke_tests_fpm.sh /srv/annotations/
-RUN composer-post
 
-RUN mkdir -p var/logs var/cache/html_purifier 
 USER root
-RUN chown www-data:www-data var/logs var/cache/html_purifier
+
+WORKDIR ${PROJECT_FOLDER}
+RUN mkdir -p var/logs var/cache/html_purifier && \
+    chown --recursive elife:elife . && \
+    chown www-data:www-data var/logs var/cache/html_purifier
+
+COPY --chown=elife:elife smoke_tests_fpm.sh .
+COPY --chown=elife:elife web/ web/
+COPY --chown=elife:elife config/ config/
+COPY --from=build --chown=elife:elife /app/vendor/ vendor/
+COPY --from=build --chown=elife:elife /app/src/ src/
```

Comment:
some symlinking necessary for backward compatibility? https://github.com/elifesciences/annotations-formula/blob/cf123e78b2f9eb020759bb4ec865353550cbc10e/salt/annotations/config/home-deployuser-annotations-docker-compose.yml#L22-L23

Reply:
It's still `/srv/annotations`?